techniques/system/rudderSlapd/1.0/main.st

```diff
@@ -0,0 +1,47 @@
+bundle common rudder_slapd {
+  vars:
+      "service_name" string => "rudder-slapd";
+      "user_name"    string => "rudder-slapd";
+      "group_name"   string => "rudder-slapd";
+
+      "config_file"  string => "${g.rudder_base}/etc/openldap/slapd.conf";
+}
+bundle agent system_rudder_slapd_configuration {
+  vars:
+
+      "slapd_password" string => "&RUDDER_LDAP_PASSWORD&";
+
+      "technique_name" string => "Rudder slapd";
+      "trackingkey"    string => "&TRACKINGKEY&";
+
+  classes:
+      "rudder_server_system_restart_slapd" expression => "rudder_slapd_ldap_password_repaired",
+                                                scope => "namespace";
+      "rudder_server_system_restart_jetty" expression => "rudder_slapd_ldap_password_repaired",
+                                                scope => "namespace";
+
+  files:
+      "${rudder_slapd.config_file}"
+        edit_line => set_rudder_slapd_password("${slapd_password}"),
+        classes   => classes_generic("rudder_slapd_ldap_password"),
+        comment   => "Verifying the Rudder LDAP file password and user";
+
+  methods:
+      "any" usebundle => _method_reporting_context("Rudder slapd configuration", "Check Rudder ldap permissions");
+      "any" usebundle => permissions("${rudder_slapd.config_file}", "640", "root", "${rudder_slapd.group_name}");
```

Comment:
why is it better to do the permissions there, rather than adding a mog("640", "root", "${rudder_slapd.group_name}"); in the files promise?

Reply:
I tried to migrate as much promises as possible to ncf. This make it easier to read for Rudder maintainers and users, Also, I personally don't think that hiding changes under unified reporting is a good thing.

Reply:
i'm not sure it's hiding much, but ok. In this case, if permissions were incorrect, we need to restart slapd as well, as config file may not have been readable